src/commands/deleteResourceGroup/v2/deleteResourceGroupV2.ts

```diff
@@ -3,72 +3,110 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { AzureWizard, IActionContext, UserCancelledError } from '@microsoft/vscode-azext-utils';
+import { ResourceGroup } from '@azure/arm-resources';
+import { uiUtils } from '@microsoft/vscode-azext-azureutils';
+import { AzureWizard, IActionContext, IAzureQuickPickItem, nonNullProp, subscriptionExperience, UserCancelledError } from '@microsoft/vscode-azext-utils';
+import { AzureResource, AzureSubscription } from '@microsoft/vscode-azext-utils/hostapi.v2';
+import { DefaultAzureResourceProvider } from '../../../api/DefaultAzureResourceProvider';
 import { ext } from '../../../extensionVariables';
 import { GroupingItem } from '../../../tree/azure/GroupingItem';
 import { createActivityContext } from '../../../utils/activityUtils';
+import { createResourceClient } from '../../../utils/azureClients';
 import { localize } from '../../../utils/localize';
 import { settingUtils } from '../../../utils/settingUtils';
+import { createSubscriptionContext } from '../../../utils/v2/credentialsUtils';
 import { DeleteResourceGroupContext } from '../DeleteResourceGroupContext';
 import { DeleteResourceGroupStep } from '../DeleteResourceGroupStep';
 
 export async function deleteResourceGroupV2(context: IActionContext, primaryNode?: GroupingItem, selectedNodes?: GroupingItem[]): Promise<void> {
+
+    // unset nodes that are not resource groups
+    selectedNodes = selectedNodes?.filter(n => !!n.resourceGroup);
+    if (!primaryNode?.resourceGroup) {
+        primaryNode = undefined;
+    }
+
+    let subscription: AzureSubscription;
+    let resourceGroupsToDelete: AzureResource[] = [];
+
     if (!selectedNodes) {
         if (primaryNode) {
             selectedNodes = [primaryNode];
+            subscription = primaryNode.subscription;
+            resourceGroupsToDelete = [nonNullProp(primaryNode, 'resourceGroup')];
         } else {
-            selectedNodes = [];
-            // TODO: Deal with picker.
-            // const subscription: SubscriptionTreeItem = await ext.appResourceTree.showTreeItemPicker(SubscriptionTreeItem.contextValue, context);
-            // selectedNodes = await subscription.pickResourceGroup(context, {
-            //    canPickMany: true,
-            //    placeholder: localize('selectResourceGroupToDelete', 'Select resource group(s) to delete'),
-            //});
+            ({ subscription, resourceGroupsToDelete } = await pickResourceGroups(context));
         }
     } else {
         selectedNodes = selectedNodes.filter(n => !!n.resourceGroup);
```

Comment:
Do `selectedNodes` also include the `primaryNode`? I could see it technically being "selected".

Reply:
I'll test that, regardless it's very confusing 😄

Reply:
So yeah. The primary node is whatever node has been right clicked to bring up the context menu. And it is included in the selected nodes array.